tools/build/src/mocha-reporter/index.ts

```diff
@@ -6,12 +6,33 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 /* eslint-disable no-console */
 const debugLeaks = process.env.DEBUG_LEAKS;
-if (debugLeaks)
+let asyncResourceStats: Map<number, any>;
+if (debugLeaks) {
   require("wtfnode");
+  asyncResourceStats = new Map<number, any>();
+  setupAsyncHooks();
+}
 
-import * as path from "path";
+function setupAsyncHooks() {
+  const async_hooks = require("node:async_hooks");
+
+  const init = (asyncId: number, type: string, triggerAsyncId: number, _resource: any) => {
+    const eid = async_hooks.executionAsyncId(); // (An executionAsyncId() of 0 means that it is being executed from C++ with no JavaScript stack above it.)
+    const stack = new Error().stack;
+    asyncResourceStats.set(asyncId, {type, eid, triggerAsyncId, initStack: stack});
+  };
+  const destroy = (asyncId: number) => {
+    if (asyncResourceStats.get(asyncId) === undefined) {
+      return;
+    }
+    asyncResourceStats.delete(asyncId);
+  };
 
+  const asyncHook = async_hooks.createHook({init, destroy});
+  asyncHook.enable();
+}
 const fs = require("fs-extra");
+import * as path from "path";
```

Comment:
feels weird using both import and require, can we consolidate and just use require here?

Reply:
Changed it